src/client/api/json_parser/payloads.c

```diff
@@ -422,12 +435,22 @@ int json_tagged_deserialize(cJSON* payload, tagged_data_t** tagged_data) {
 
   // create a new tagged data
   if (cJSON_IsString(json_tag) && cJSON_IsString(json_data)) {
-    *tagged_data =
-        tagged_data_create(json_tag->valuestring, (byte_t*)json_data->valuestring, strlen(json_data->valuestring));
+    char* tag_str = malloc((strlen(json_tag->valuestring) / 2));
```

Comment:
* the max length of the tag is 64 bytes, use static buffer instead.
* `tagged_data_t` stores tag and data in binary form, therefore tag and data should convert to binary via `hex_2_bin` method.

Reply:
Used static buf and hex_2_bin.